packages/algob/src/internal/deployer.ts

```diff
@@ -275,13 +282,47 @@ export class DeployerDeployMode extends DeployerBasicMode implements AlgobDeploy
     return lsigInfo;
   }
 
+  /**
+   * Description: function to deploy stateful smart contracts
+   * @param approvalProgram filename which has approval program
+   * @param clearProgram filename which has clear program
+   * @param flags SSCDeploymentFlags
+   * @param payFlags Transaction Params
+   */
+  async deploySSC (
+    approvalProgram: string,
+    clearProgram: string,
+    flags: SSCDeploymentFlags,
+    payFlags: TxParams): Promise<SSCInfo> {
+    const name = approvalProgram + "-" + clearProgram;
+    this.assertNoAsset(name);
+    let sscInfo = {} as any;
+    try {
+      sscInfo = await this.algoOp.deploySSC(
+        approvalProgram, clearProgram, flags, payFlags, this.txWriter);
+    } catch (error) {
+      persistCheckpoint(this.txWriter.scriptName, this.cpData.strippedCP);
+
+      console.log(error);
+      throw error;
+    }
+
+    this.cpData.registerSSC(this.networkName, name, sscInfo);
+
+    return sscInfo;
+  }
+
   async optInToASA (name: string, accountName: string, flags: TxParams): Promise<void> {
     await this.algoOp.optInToASA(
       name,
       this._getASAInfo(name).assetIndex,
       this._getAccount(accountName),
       flags);
   }
+
+  async OptInToSSC (sender: Account, index: number, payFlags: TxParams): Promise<void> {
```

Comment:
Since we have two opt-in functions now, `OptInToSSC` (application call type - OptIn) and `optInToASA` (0 asset transfer to itself). Should we add a comment to this function as well ?

Reply:
yes